### PR TITLE
Added new endpoint login and JWT to confirm users

### PR DESCRIPTION
**Pull request by Dima23-ops5**

### CODE REVIEW

File: src/main/java/mate/academy/springbootintro/config/OpenApiConfig.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    @Bean
+    public OpenAPI customOpenApi() {
+        return new OpenAPI()
+                .components(new Components().addSecuritySchemes("BearerAuth",
+                        new SecurityScheme()
+                                .type(SecurityScheme.Type.HTTP)
+                                .scheme("bearer")
+                                .bearerFormat("JWT")))
+                .addSecurityItem(new SecurityRequirement().addList("BearerAuth"));
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/springbootintro/config/OpenApiConfig.java]: Let`s save it to constants

File: src/main/java/mate/academy/springbootintro/config/SecurityConfig.java [hunk 2]
```diff
@@ -1,21 +1,27 @@
-package mate.academy.springbootintro.security;
+package mate.academy.springbootintro.config;
 
-import static org.springframework.security.config.Customizer.withDefaults;
+import  static org.springframework.security.config.Customizer.withDefaults;
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/springbootintro/config/SecurityConfig.java]:
```suggestion
import static org.springframework.security.config.Customizer.withDefaults;
```

Comment (MentorMateAcademy) [reply src/main/java/mate/academy/springbootintro/config/SecurityConfig.java]: import formatting

File: src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java]:
```suggestion
```

File: src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    private final JwtUtil jwtUtil;
+    private final CustomUserDetailsService userDetailsService;
+    @Override
+    protected void doFilterInternal(
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java]:
```suggestion
    private final CustomUserDetailsService userDetailsService;
    
    @Override
    protected void doFilterInternal(
```

File: src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+
+        }
+       filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (bearerToken != null && bearerToken.startsWith("Bearer")) {
+            return bearerToken.substring(7);
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java]: save to constants

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/springbootintro/config/SecurityConfig.java (hunk 2 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/springbootintro/config/SecurityConfig.java]: not fixed

File: src/main/java/mate/academy/springbootintro/exeption/LoginException.java [hunk 6]
```diff
@@ -0,0 +1,7 @@
+package mate.academy.springbootintro.exeption;
+
+public class LoginException extends RuntimeException {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/exeption/LoginException.java]: You don't use this exception, remove it

File: src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+                            null,
+                            userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+
+        }
+       filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(AUTHORIZATION);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java]: Use Spring constant
```suggestion
        String bearerToken = request.getHeader(HttpHeaders.AUTHORIZATION);
```

File: src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private static final String AUTHORIZATION = "Authorization";
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java]:
```suggestion
```

File: src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    private static final String BEARER = "Bearer";
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java]:
```suggestion
    private static final String TOKEN_HEADER = "Bearer ";
```

File: src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    private static final int DEFAULT_CAPACITY = 7;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java]:
```suggestion
```

See comment below

File: src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+
+        }
+       filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(AUTHORIZATION);
+        if (bearerToken != null && bearerToken.startsWith(BEARER)) {
+            return bearerToken.substring(DEFAULT_CAPACITY);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springbootintro/security/JwtAuthenticationFilter.java]:
```suggestion
            return bearerToken.substring(TOKEN_HEADER.length());
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

**Review by sarakhmen — APPROVED** [review]

**Pull request closed without merging**